test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
 });
 
 describe('client receiving message', () => {
-    it('should call commandController.listTasksForUser when command is "!TASKS LIST"', done => {
-        expect.assertions(3);
-
-        const client = setupClient();
-
-        const returnedTasks = new RichEmbed();
-
-        const commandControllerMock = new CommandControllerMockBuilder()
-            .withListTasksForUser(jest.fn(() => Promise.resolve(returnedTasks)))
-            .build();
-
-        const discordController = new DiscordControllerBuilder()
-            .withCommandController(commandControllerMock)
-            .withClient(client)
-            .build();
-
-        const sentMessage = {
-            edit: jest.fn(async value => {
-                expect(commandControllerMock.listTasksForUser).toBeCalledWith(message.author);
-                expect(value).toBe(returnedTasks);
-                expect(message.channel.startTyping).toBeCalled();
-                done();
-            })
-        };
-
-        const message = {
-            content: '!TASKS LIST',
-            author: 'author',
-            channel: {
-                send: jest.fn(async () => sentMessage),
-                startTyping: jest.fn(() => Promise.resolve()),
-                stopTyping: jest.fn(() => Promise.resolve()),
-            }
-        };
-
-        client.emit('message', message);
+    describe('when message is "!tasks create"', () => {
+        it('should call commandController.createTask when command'
+            + ' is sent from project channel', () => {
+            expect.assertions(3);
+
+            const firstMessage = 'Creating task...';
+            let messagesSent = 0;
+
+            const client = setupClient();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .withCommandController(commandControllerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks create new task')
+                .withSend(jest.fn(async value => {
+                    messagesSent++;
+
+                    if (messagesSent === 1) {
```

Comment:
I think this `if (messagesSent === 1)` is unnecessary, since it should only be called once anyway. It would only be useful if we threw an error if messagesSent was not the expected value, but we do that anyway later in the test.